core-container/entry.sh

```diff
@@ -23,145 +23,133 @@ function echoerr() {
-    kubectl create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig="kcpsecret"    
-    rm kcpsecret
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig=$kcp_kubeconfig   
 
     echo "Delete the kcp provider object if it already exists, and then create it."
-    if ! kubectl delete spaceproviderdesc pkcp ; then
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete spaceproviderdesc $PROVIDER_NAME ; then
         echo "Nothing to delete."
     fi
-    kubectl create -f - <<EOF
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkcp
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kcp"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: kcpsec
 EOF
-
-    if [ "$SPACE_PROVIDER" == "kubeflex" ]; then
-        echo "Waiting for the kubeflex provider to be ready... this may take a while."
-        (
-            until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
-                sleep 10
-            done
-        )
+    echo "Waiting for spaceprovider to reach the Ready phase."
+    kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} wait --for=jsonpath='{.status.Phase}'=Ready spaceproviderdesc $PROVIDER_NAME
+}
  
-        echo "Delete the kubeflex provider object if it already exists, and then create it."
-        if ! kubectl delete spaceproviderdesc pkflex ; then
-            echo "Nothing to delete."
-        fi
-        kubectl create -f - <<EOF
+function create_kubeflex_provider() {
+    echo "Waiting for the kubeflex provider to be ready... this may take a while."
+    (
+        until [ "$(kubectl --kubeconfig $kubestellar_kubeconfig get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
+            sleep 10
+        done
+    )
+
+    echo "Delete the kubeflex provider object if it already exists, and then create it."
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete spaceproviderdesc $PROVIDER_NAME ; then
+        echo "Nothing to delete."
+    fi
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkflex
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kubeflex"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: corecluster
 EOF
-    fi
-
-    KUBECONFIG=$oldKubeconfig
+    echo "Waiting for spaceprovider to reach the Ready phase."
+    kubectl --kubeconfig ${SPACE_MANAGER_KUBECONFIG} wait --for=jsonpath='{.status.Phase}'=Ready spaceproviderdesc $PROVIDER_NAME
 }
 
 
-function wait_kcp_ready() {
-    echo "Waiting for kcp to be ready... this may take a while."
+function create_spaceprovider() {
```

Comment:
Like the create_xxx_provider functions it calls, this function is not actually _creating_ the _provider_.